core/trino-main/src/main/java/io/trino/server/security/AbstractBearerAuthenticator.java

```diff
@@ -52,7 +53,9 @@ public Identity authenticate(ContainerRequestContext request, String token)
                 throw needAuthentication(request, "Invalid credentials");
             }
 
-            String authenticatedUser = userMapping.mapUser(principal.get().getName());
+            String username = principal.get().getName();
+            String authenticatedUser = userMapping.mapUser(username);
+            checkAndRewriteUserHeaderToMappedUserWhenRequired(username, request.getHeaders(), authenticatedUser);
```

Comment:
The rewrite hack is supposed to only be applied to username+password authentication, because the user name is both the trino run as user and the authentication user.  In all other cases, it is expected that the user will either pass the correct user they will be mapped to, or that the client is configured to not send the username at all.